xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -164,19 +196,36 @@ public WeightedChildLbState(Object key, LoadBalancerProvider policyProvider, Obj
     }
 
     private double getWeight() {
-      if (config == null) {
-        return 0;
-      }
-      long now = ticker.nanoTime();
-      if (now - lastUpdated >= config.weightExpirationPeriodNanos) {
-        nonEmptySince = infTime;
-        return 0;
-      } else if (now - nonEmptySince < config.blackoutPeriodNanos
-          && config.blackoutPeriodNanos > 0) {
-        return 0;
-      } else {
-        return weight;
+      double weightToReturn = 0;
+      if (config != null) {
+        long now = ticker.nanoTime();
+        if (weightIsStale(now)) {
+          nonEmptySince = infTime;
+          // TODO: add target and locality labels once available
+          endpointWeightStaleCounter.add(1, ImmutableList.of(), ImmutableList.of());
```

Comment:
Looks like you need to pass values in these lists.

Reply:
These are the values we decided to leave empty for now until we plumb through target and locality.

Reply:
Ok, understood - they need to use empty strings like in the other instances.